libopae/api-shell.c

```diff
@@ -164,1123 +139,34 @@ fpga_result fpgaGetProperties(fpga_token token, fpga_properties *prop)
-			FPGA_NOT_SUPPORTED);
-
-		return wrapped_properties->adapter_table
-			->fpgaPropertiesSetAcceleratorState(
-				wrapped_properties->opae_properties, state);
-	}
-
-	return opae_properties_set_accelerator_state(
-		wrapped_properties->opae_properties, state);
-}
-
-fpga_result fpgaPropertiesGetObjectID(const fpga_properties prop,
-				      uint64_t *object_id)
-{
-	opae_wrapped_properties *wrapped_properties =
-		opae_validate_wrapped_properties(prop);
-
-	ASSERT_NOT_NULL(wrapped_properties);
-	ASSERT_NOT_NULL(object_id);
-
-	if (wrapped_properties->adapter_table) {
-		ASSERT_NOT_NULL_RESULT(wrapped_properties->adapter_table
-					       ->fpgaPropertiesGetObjectID,
-				       FPGA_NOT_SUPPORTED);
-
-		return wrapped_properties->adapter_table
-			->fpgaPropertiesGetObjectID(
-				wrapped_properties->opae_properties, object_id);
-	}
-
-	return opae_properties_get_object_id(
-		wrapped_properties->opae_properties, object_id);
-}
-
-fpga_result fpgaPropertiesSetObjectID(const fpga_properties prop,
-				      uint64_t object_id)
-{
-	opae_wrapped_properties *wrapped_properties =
-		opae_validate_wrapped_properties(prop);
-
-	ASSERT_NOT_NULL(wrapped_properties);
-
-	if (wrapped_properties->adapter_table) {
-		ASSERT_NOT_NULL_RESULT(wrapped_properties->adapter_table
-					       ->fpgaPropertiesSetObjectID,
-				       FPGA_NOT_SUPPORTED);
-
-		return wrapped_properties->adapter_table
-			->fpgaPropertiesSetObjectID(
-				wrapped_properties->opae_properties, object_id);
-	}
-
-	return opae_properties_set_object_id(
-		wrapped_properties->opae_properties, object_id);
-}
-
-fpga_result fpgaPropertiesGetNumErrors(const fpga_properties prop,
-				       uint32_t *num_errors)
-{
-	opae_wrapped_properties *wrapped_properties =
-		opae_validate_wrapped_properties(prop);
-
-	ASSERT_NOT_NULL(wrapped_properties);
-	ASSERT_NOT_NULL(num_errors);
-
-	if (wrapped_properties->adapter_table) {
-		ASSERT_NOT_NULL_RESULT(wrapped_properties->adapter_table
-					       ->fpgaPropertiesGetNumErrors,
-				       FPGA_NOT_SUPPORTED);
-
-		return wrapped_properties->adapter_table
-			->fpgaPropertiesGetNumErrors(
-				wrapped_properties->opae_properties,
-				num_errors);
-	}
-
-	return opae_properties_get_num_errors(
-		wrapped_properties->opae_properties, num_errors);
-}
-
-fpga_result fpgaPropertiesSetNumErrors(const fpga_properties prop,
-				       uint32_t num_errors)
-{
-	opae_wrapped_properties *wrapped_properties =
-		opae_validate_wrapped_properties(prop);
-
-	ASSERT_NOT_NULL(wrapped_properties);
-
-	if (wrapped_properties->adapter_table) {
-		ASSERT_NOT_NULL_RESULT(wrapped_properties->adapter_table
-					       ->fpgaPropertiesSetNumErrors,
-				       FPGA_NOT_SUPPORTED);
-
-		return wrapped_properties->adapter_table
-			->fpgaPropertiesSetNumErrors(
-				wrapped_properties->opae_properties,
-				num_errors);
-	}
-
-	return opae_properties_set_num_errors(
-		wrapped_properties->opae_properties, num_errors);
+		wrapped_token->opae_token, prop);
```

Comment:
Don't forget to wrap the parent token if present in the properties.

Reply:
But I don't see the change here ... do we need to do it here too?

Reply:
thanks!